graphs/util/gpml_util_interface.py

```diff
@@ -0,0 +1,345 @@
+
+import graphspace_interface as interface
+import networkx as nx
+import json
+
+
+def parse_gpml(graph_json, title):
+    parse_json = {"graph": {"edges": [], "nodes": []}, "metadata": {}}
+    G = nx.DiGraph(directed=True)
+    # Use xml parser to parse the GPML file
+    import xml.etree.ElementTree as ET
+    tree = ET.fromstring(graph_json)
+    # The current GPML format
+    GPML = '{http://pathvisio.org/GPML/2013a}'
+    # Keep the overall count of nodes and create unique ids for
+    # datanodes, shape, labels.
+    # IMPORTANT!!!
+    count = 0
+    # Parsing the GPML file for Shape elements
+    shapes = tree.findall(GPML+'Shape')
+    G, parse_json, count = parse_shapes(G, parse_json, count, shapes)
+    # Find all the DataNodes in the GPML file which is used to draw various
+    # the various nodes of the graph
+    datanodes = tree.findall(GPML+'DataNode')
+    G, parse_json, count = parse_datanodes(G, parse_json, count, datanodes)
+    # Parsing the GPML file for Shape elements
+    labels = tree.findall(GPML+'Label')
+    G, parse_json, count = parse_labels(G, parse_json, count, labels)
+    # This is a crude version of getting interactions (edges) working
+    # Look for all the interactions
+    interactions = tree.findall(GPML+'Interaction')
+    G, parse_json, count = parse_interactions(G, parse_json, count, interactions)
+    print G.nodes(data=True)
+    layout = set_gpml_layout(G, parse_json)
+
+    # To add nodes to the group we parse through the group
+    # elements of the GPML file.
+    groups = tree.findall(GPML+'Group')
+    G, parse_json = parse_groups(G, parse_json, groups)
+    parse_json['metadata']['name'] = tree.attrib['Name'] or "GPML"
+
+    # No tags or description since CYJS doesn't give me any
+    parse_json['metadata']['tags'] = []
+    parse_json['metadata']['description'] = ""
+    title = title or parse_json['metadata']['name']
+    
+    metadata = {'description':'example1','title':'Example 1 Graph','tags':[]}
+    out = interface.convertNXToDict(G, metadata=metadata)
+
+    return out, parse_json, layout, title
+
+
+def parse_shapes(G, parse_json, count, shapes):
+    # Create a list of details of all the shape
+    shape_details = []
+    for shape in shapes:
```

Comment:
Are we calling them details or attributes? Is "detail" a concept used in GPML or is "attribute" the right word?